chatcommands.py

```diff
@@ -1141,128 +1143,68 @@ def report(msg, args):
-        add_or_update_multiple_reporter(msg.owner.id, msg._client.host, time.time())
-
-    if len(output) > 0:
-        return os.linesep.join(output)
-
-
-# noinspection PyIncorrectDocstring,PyUnusedLocal
-@command(str, whole_msg=True)
-def scan(msg, args):
-    """
-    Force Smokey to scan a post even if it has no recent activity
-    :param msg:
-    :param url:
-    :return str:
-    """
-    crn, wait = can_report_now(msg.owner.id, msg._client.host)
-    if not crn:
-        raise CmdException("You can execute the !!/scan command again in {1} seconds. "
-                           "To avoid one user sending lots of reports in a few commands and "
-                           "slowing SmokeDetector down due to rate-limiting, you have to "
-                           "wait 30 seconds after you've scanned multiple posts in "
-                           "one go.".format(wait))
-
-    urls = args.split()
-
-    if len(urls) > 5:
-        raise CmdException("To avoid SmokeDetector reporting posts too slowly, you can "
-                           "scan at most 5 posts at a time. This is to avoid "
-                           "SmokeDetector's chat messages getting rate-limited too much, "
-                           "which would slow down reports.")
-
-    response = []
+        if fetch_post_id_and_site_from_url(url)[2] == "answer":
+            parent_data = api_get_post("https://{}/q/{}".format(post.post_site, post_data.question_id))
+            post._is_answer = True
+            post._parent = Post(api_response=parent_data.as_dict)
 
-    for index, url in enumerate(urls):
-        post_data = api_get_post(rebuild_str(url))
+        scan_spam, scan_reasons, scan_why = check_if_spam(post)  # Scan it first
 
-        if post_data is None:
-            response.append((index, "That does not look like a valid post URL."))
-            continue
+        # Here's where it starts to be different
 
-        if post_data is False:
-            response.append((index, "Cannot find data for this post in the API. "
-                                    "It may have already been deleted."))
+        # If alias_used == "report-force" then jump to the next block
+        if scan_spam and alias_used in {"scan", "report"}:
+            handle_spam(post=post, reasons=scan_reasons, why=scan_why + "\nManually triggered scan")
             continue
 
-        # Update url to be consistent with other code
-        url = to_protocol_relative(post_data.post_url)
-        post = Post(api_response=post_data.as_dict)
-
-        if has_already_been_posted(post_data.site, post_data.post_id, post_data.title) \
-                and not is_false_positive((post_data.post_id, post_data.site)):
-            # Don't re-report if the post wasn't marked as a false positive. If it was marked as a false positive,
-            # this force scan might be attempting to correct that/fix a mistake/etc.
+        # scan_spam == False
+        if alias_used in {"report", "report-force"}:
```

Comment:
Aha, so we only wind up submitting a custom report if we use one of these two aliases, so `!!/scan` never submits a manual report. :)

Reply:
@spacemonaut I just pushed [a commit][1] that enabled `scan` with custom reasons. The report should look like this now:

```
Post manually scanned by *iBug* in room *Charcoal HQ* with reason: *\_(ツ)_/*.

Body: Position 123-456: blah blah blah
```

[1]: https://github.com/Charcoal-SE/SmokeDetector/commit/50af479ba164aea606191ca4df7078f82585a2d6